### PR TITLE
Refactor tokens processor

### PR DESCRIPTION
**Pull request by miiu96**
- Refactored the tokens processor: instead of extract information from transactions data fields now we are using logs and events

### CODE REVIEW

File: process/logsevents/interface.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ type argsProcessEvent struct {
 	tokens           data.TokensHandler
 	tagsCount        data.CountTags
 	timestamp        uint64
+	tokensInfo       *[]*data.TokenInfo
```
Comment (iulianpascalau) [inline process/logsevents/interface.go:17]: what is this?

Comment (miiu96) [reply process/logsevents/interface.go]: refactored

File: process/logsevents/issueESDTProcessor.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	topics := args.event.GetTopics()
+	if len(topics) < numIssueLogTopics {
+		return argOutputProcessEvent{
+			processed: true,
+		}
+	}
+
+	// TOPICS contains
```
Comment (bogdan-rosianu) [inline process/logsevents/issueESDTProcessor.go:49]: `// topics slice contains:` *

File: process/logsevents/issueESDTProcessor.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	numIssueLogTopics = 4
+
+	issueFungibleESDTFunc     = "issue"
+	issueSemiFungibleESDTFunc = "issueSemiFungible"
+	issueNonFungibleESDTFunc  = "issueNonFungible"
+	registerMetaESDTFunc      = "registerMetaESDT"
+	changeSFTToMetaESDTFunc   = "changeSFTToMetaESDT"
+)
+
+type issueESDTProcessor struct {
```
Comment (bogdan-rosianu) [inline process/logsevents/issueESDTProcessor.go:19]: rename file, struct to `esdtIssueProcessor` or `esdtIssuanceProcessor`

File: process/logsevents/interface.go [hunk 4]
```diff
@@ -16,6 +16,13 @@ type argsProcessEvent struct {
 	timestamp        uint64
 }
 
+type argOutputProcessEvent struct {
+	identifier string
+	value      string
+	processed  bool
+	tokenInfo  *data.TokenInfo
```
[the remark this answers is not in the document]
Comment (miiu96) [reply process/logsevents/interface.go:23]: done.

File: process/logsevents/esdtIssueProcessor_test.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	"github.com/ElrondNetwork/elastic-indexer-go/data"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIssueESDTProcessor(t *testing.T) {
+	t.Parallel()
+
+	issueESDTProc := newESDTIssueProcessor()
```
Comment (bogdan-rosianu) [inline process/logsevents/esdtIssueProcessor_test.go:16]: esdtIssueProc

File: process/logsevents/logsAndEventsProcessor.go [hunk 6]
```diff
@@ -38,13 +38,15 @@ func NewLogsAndEventsProcessor(
 	nftsProc := newNFTsProcessor(shardCoordinator, pubKeyConverter, marshalizer)
 	fungibleProc := newFungibleESDTProcessor(pubKeyConverter, shardCoordinator)
 	scDeploysProc := newSCDeploysProcessor(pubKeyConverter)
+	issueESDTProc := newESDTIssueProcessor()
```
Comment (bogdan-rosianu) [inline process/logsevents/logsAndEventsProcessor.go:41]: esdtIssueProc

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**